sdk/src/asset_handlers/zip_io.rs

```diff
@@ -0,0 +1,380 @@
+            "xlsm",
+            "application/vnd.ms-excel.sheet.macroenabled.12",
+            "pptm",
+            "application/vnd.ms-powerpoint.presentation.macroenabled.12",
+            // Open Document
+            "odt",
+            "application/vnd.oasis.opendocument.text",
+            "ods",
+            "application/vnd.oasis.opendocument.spreadsheet",
+            "odp",
+            "application/vnd.oasis.opendocument.presentation",
+            "odg",
+            "application/vnd.oasis.opendocument.graphics",
+            "ott",
+            "application/vnd.oasis.opendocument.text-template",
+            "ots",
+            "application/vnd.oasis.opendocument.spreadsheet-template",
+            "otp",
+            "application/vnd.oasis.opendocument.presentation-template",
+            "otg",
+            "application/vnd.oasis.opendocument.graphics-template",
+            // OpenXPS
+            "oxps",
+            "application/oxps",
+        ]
+    }
+}
+
+impl ZipIO {
+    fn reader<'a>(
+        &self,
+        input_stream: &'a mut dyn CAIRead,
+    ) -> ZipResult<ZipArchive<CAIReadWrapper<'a>>> {
+        ZipArchive::new(CAIReadWrapper {
+            reader: input_stream,
+        })
+    }
+
+    fn writer<'a>(
+        &self,
+        input_stream: &'a mut dyn CAIRead,
+        output_stream: &'a mut dyn CAIReadWrite,
+    ) -> ZipResult<ZipWriter<CAIReadWriteWrapper<'a>>> {
+        input_stream.rewind()?;
+        io::copy(input_stream, output_stream)?;
+
+        ZipWriter::new_append(CAIReadWriteWrapper {
+            reader_writer: output_stream,
+        })
+    }
+}
+
+// TODO: probably doesn't need to return a vec
+pub fn central_directory_inclusions<R>(reader: &mut R) -> Result<Vec<HashRange>>
+where
+    R: Read + Seek + ?Sized,
+{
+    let _reader = ZipArchive::new(reader).map_err(|_| Error::JumbfNotFound)?;
+
+    // TODO: https://github.com/zip-rs/zip2/issues/209
+
+    todo!()
+}
+
+pub fn uri_maps<R>(stream: &mut R) -> Result<Vec<UriHashedDataMap>>
+where
+    R: Read + Seek + ?Sized,
+{
+    let mut reader = ZipArchive::new(stream).map_err(|_| Error::JumbfNotFound)?;
+
+    let mut uri_maps = Vec::new();
+    let file_names: Vec<String> = reader.file_names().map(|n| n.to_owned()).collect();
+    for file_name in file_names {
+        let file = reader
+            .by_name(&file_name)
+            .map_err(|_| Error::JumbfNotFound)?;
+
+        if !file.is_dir() {
+            uri_maps.push(UriHashedDataMap {
+                // TODO: temp unwrap
+                #[allow(clippy::unwrap_used)]
+                uri: file.enclosed_name().unwrap(),
+                hash: Vec::new(),
+                size: Some((file.data_start() + file.compressed_size()) - file.header_start()),
+                dc_format: None,
+                data_types: None,
+            });
+        }
+    }
+
+    Ok(uri_maps)
+}
+
+pub fn uri_inclusions<R>(stream: &mut R, uri_maps: &[UriHashedDataMap]) -> Result<Vec<HashRange>>
```

Comment:
The hash is per each individual files compressed content.  I don't this this function is needed.  See my suggestions for uri_maps function.